contracts/OlasMech.sol

```diff
@@ -264,21 +274,17 @@ abstract contract OlasMech is Mech, IErrorsMech, ImmutableStorage {
 
     /// @dev Delivers signed requests.
     /// @param requester Requester address.
-    /// @param requestDatas Corresponding set of self-descriptive opaque request data-blobs.
-    /// @param signatures Corresponding set of signatures.
+    /// @param deliverWithSignatures Set of DeliverWithSignature structs.
     /// @param deliveryRates Corresponding set of actual charged delivery rates for each request.
-    /// @param deliveryDatas Corresponding set of self-descriptive opaque delivery data-blobs.
     /// @param paymentData Additional payment-related request data, if applicable.
     function deliverMarketplaceWithSignatures(
         address requester,
-        bytes[] memory requestDatas,
-        bytes[] memory signatures,
-        bytes[] memory deliveryDatas,
-        uint256[] memory deliveryRates,
+        DeliverWithSignature[] calldata deliverWithSignatures,
+        uint256[] calldata deliveryRates,
```

Comment:
why not also including the `deliveryRates` in `deliverWithSignatures`?

Reply:
The `deliveryRates`  array is going to be then used as an input parameter in `IBalanceTracker`.